docs/source/scheduling-policies.rst

```diff
@@ -82,25 +82,44 @@ where can the task run the soonest, considering both data transfer and worker bu
 
 Tasks that don't meet the root-ish criteria described above are selected as follows:
 
+First, we identify the pool of viable workers:
+
 1.  If the task has no dependencies and no restrictions, then we find the
     least-occupied worker.
 2.  Otherwise, if a task has user-provided restrictions (for example it must
     run on a machine with a GPU) then we restrict the available pool of workers
-    to just that set, otherwise we consider all workers.
-3.  From among this pool of workers, we determine the workers to whom the least
-    amount of data would need to be transferred.
-4.  We break ties by choosing the worker that currently has the fewest tasks,
-    counting both those tasks in memory and those tasks processing currently.
+    to just that set. Otherwise, we consider all workers.
+3.  We restrict the above set to just workers that hold at least one dependency
+    of the task.
+
+From among this pool of workers, we then determine the worker where we think the task will start
+running the soonest, using :method:`Scheduler.worker_objective`. For each worker:
+
+1.  We consider the estimated runtime of other tasks already queued on that worker.
+    Then, we add how long it will take to transfer any dependencies to that worker that
+    it doesn't already have, based on their size, in bytes, and the measured network
+    bandwith between workers. (This does *not* consider serialization time, nor whether
+    the data is spilled to disk versus in memory on the worker.) In practice, the
+    queue-wait-time (known as *occupancy*) usually dominates, so data will usually be
+    transferred to a different worker if it means the task can start any sooner.
+2.  It's possible for ties to occur with the "start soonest" metric, though uncommon
+    when all workers are busy. We break ties by choosing the worker that has the
+    fewest number of bytes of Dask data stored (including spilled data). Note that
+    this is not the same as :ref:`managed memory <memtypes>`, nor the total memory
+    used by the process.
```

Comment:
```suggestion
    fewest number of bytes of Dask data stored (including spilled data). Note that
    this is the same as :ref:`managed <memtypes>` plus :ref:`unmanaged <memtypes>`
    memory, not the :ref:`process <memtypes>` memory.
```

Reply:
Sorry, typo - I meant to write plus spilled.
In the referenced high level document as well as in the GUI, "managed" memory maps to ``MemoryState.managed_in_memory``, not to ``MemoryState.managed`` which also includes spilled. This is a deliberate misalignment to reduce confusion in newer users.

Reply:
``MemoryState.managed_in_memory = min(WorkerState.nbytes, MemoryState.process)`` so in most cases they are the same.